### DISCUSSION TITLE
Android push notifications

### DISCUSSION THREAD

**Discussion by mfnboer** [bluesky-social/atproto#1914] [How to do it?] [answered]
Hi,
I am trying to get push notifications from Bluesky sent to my own Android app, but I get stuck and need some help.
I send app.bsky.notification.registerPush with the following parameters:

- serviceDid = "did:web:api.bsky.app"
- token = <Firebase FCM token that I get from FirebaseMessaging.GetToken()>
- platform = "android"
- appId = "com.gmail.mfnboer.skywalker"

I have tested with Google Firebase Console that I can send a push notification to the token. That works.
However, from Bluesky I never get a push notification.
Can you tell me what I am doing wrong?

Cheers,
Michel

**Comment by mfnboer** [comment]
No one?
An answer saying that Bluesky will not send push notifications to 3rd parties is also fine. Well, not fine, but then I know I can stop trying.

**Comment by seboslaw** [comment, marked as answer]
Bluesky will not send push notifications to 3rd parties. You have to setup your own backend to do that. The registerPush API call will just sign your request and forward it to the Service you provide in the payload.

**Reply by mfnboer** [reply]
I see. That makes sense. Thanks for your explanation.
Would be good to add that as a comment in the lexicon.

**Reply by pfrazee** [reply]
Sorry for the lack of a response (I'm just now getting through my inbox)

@seboslaw is correct though we're looking into whether we can get 3rd party support added.

**Reply by mfnboer** [reply]
Hi Paul, thanks for your response! Would be cool if that would be possible in the future.

**Comment by harrymash2006** [comment]
How does that actually work? I try to call the API with this but it fails with 500. How to code the backend to receive the push from bluesky?

curl --location 'https://bsky.social/xrpc/app.bsky.notification.registerPush' \
--header 'Accept-Language: en-IN;q=1.0, hi-IN;q=0.9' \
--header 'Accept-Encoding: br;q=1.0, gzip;q=0.9, deflate;q=0.8' \
--header 'User-Agent: Bluestocks/1.0 (com.bluestocks.app; build:10; iOS 18.0.0) Alamofire/5.10.2' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer eyJ0eXAiOiJhdCtqd3QiLCJhbGciOiJFUzI1NksifQ.eyJzY29wZSI6ImNvbS5hdHByb3RvLmFwcFBhc3NQcml2aWxlZ2VkIiwic3ViIjoiZGlkOnBsYzp1eHEydGVjcGdkZHVheXNwYmR5YzRmcDYiLCJpYXQiOjE3Mzc3MTI5MzksImV4cCI6MTczNzcyMDEzOSwiYXVkIjoiZGlkOndlYjpzY2FseWNhcC51cy13ZXN0Lmhvc3QuYnNreS5uZXR3b3JrIn0.LVEZtP-n8XGYl8z4aiUuTxtx2HcyEM9G9SvWRDBvq6cLmEFYQksudqP_vA_fLlzbyc4Nb6shi72WIl7-srGPsA' \
--data '{
    "platform": "ios",
    "appId": "com.bluestocks.app",
    "serviceDid": "did:web:api.bsky.app",
    "token": "801516d2740675127b71725ed8d735737884726b7ebf596764162449a9d67938928b1783fb2e8d35eedab20ffcfd35d7886b758159f21f051bce151a8098cb84b3b8fd3c8a5ca450be4c212d612f935f"
}'

**Reply by harrymash2006** [reply]
@seboslaw any suggestions?

**Reply by DracoBlue** [reply]
you need to host a push gateway on your own. this is why I created https://github.com/DracoBlue/atproto-push-gateway for my app.

**Comment by DracoBlue** [comment]
For my own usecase I build https://github.com/DracoBlue/atproto-push-gateway to push to my own android/iOS app. It's based on jetstream and tries to trigger the same notifications like the social-app expects.

**Reply by OfficialBoyfriend** [reply]
Currently, it appears that only official notifications can trigger DM messages, and the same applies to push notifications based on users' mute settings.

**Reply by DracoBlue** [reply]
yep. also subscribing to posts, notification on starterpack joined and contact match are not possible outside of bsky servers.
